lib/phoenix_live_view/channel.ex

```diff
@@ -649,16 +649,16 @@ defmodule Phoenix.LiveView.Channel do
       router: router
     }
 
-    {params, parsed_uri, action} =
+    {params, parsed_uri, action, host} =
       case router && url && Utils.live_link_info!(socket, view, url) do
-        {:internal, params, action, parsed_uri} -> {params, parsed_uri, action}
-        _ -> {:not_mounted_at_router, :not_mounted_at_router, nil}
+        {:internal, params, action, parsed_uri} -> {params, parsed_uri, action, parsed_uri.host}
+        _ -> {:not_mounted_at_router, :not_mounted_at_router, nil, "www.example.com"}
```

Comment:
I think it should be `nil`. Otherwise a LiveView not mounted at the router is going to have `socket.host` equal to "www.example.com" and this can cause bugs.